nginx/etc/nginx/nginx.conf

```diff
@@ -0,0 +1,136 @@
+user nginx;
+worker_processes 4;
+pid /run/nginx.pid;
+
+events {
+    worker_connections 768;
+    # multi_accept on;
+}
+
+http {
+
+    ##
+    # Basic Settings
+    ##
+
+    sendfile on;
+    tcp_nopush on;
+    tcp_nodelay on;
+    keepalive_timeout 65;
+    types_hash_max_size 2048;
+    server_tokens off;
+
+    # server_names_hash_bucket_size 64;
+    # server_name_in_redirect off;
+
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream;
+
+    ##
+    # SSL Settings
+    ##
+
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
+    ssl_prefer_server_ciphers on;
+
+    ##
+    # Logging Settings
+    ##
+
+    log_format timed_combined '$remote_addr - $remote_user [$time_local] '
+                        '"$request" $status $body_bytes_sent $gzip_ratio '
+                        '"$http_referer" "$http_user_agent" '
+                        '$request_time $upstream_response_time';
+
+    access_log /var/log/nginx/access.log timed_combined;
+    error_log /var/log/nginx/error.log;
+
+    ##
+    # Gzip Settings
+    ##
+
+    gzip on;
+    gzip_types application/javascript text/css application/json;
+    gzip_disable "msie6";
+
+    ##
+    # Proxy Settings
+    ##
+
+    # Public VPC subnet CIDRs
+    set_real_ip_from 10.0.0.0/24;
+    set_real_ip_from 10.0.2.0/24;
+    set_real_ip_from 10.0.4.0/24;
+    set_real_ip_from 10.0.6.0/24;
+
+    # Docker bridge CIDR
+    set_real_ip_from 172.16.0.0/12;
+
+    real_ip_header X-Forwarded-For;
+
+    ##
+    # Virtual Host Configs
+    ##
+
+    server {
```

Comment:
Instead of bundling the core Nginx configuration and virtual host configurations, I split them out (see the `include` in the example above). The container image definition is then responsible for adding another `default.conf` into the `conf.d` directory:

```dockerfile
COPY srv/dist /srv/dist/
COPY etc/nginx/nginx.conf /etc/nginx/nginx.conf
COPY etc/nginx/conf.d/default.conf /etc/nginx/default.conf
```

**default.conf**

```nginx
server {
    listen 80;
    server_name localhost;
    return 301 https://$host$request_uri;
}

map $http_x_forwarded_proto $policy {
    default "";
    https   "default-src https: data: blob: 'unsafe-inline' 'unsafe-eval'";
}

upstream django-upstream {
    server django:8080;
}

server {
    listen 443 default_server;
    server_name localhost;

    # A set of recommended security headers:
    #
    #   https://scotthelme.co.uk/hardening-your-http-response-headers/
    #
    add_header Strict-Transport-Security "max-age=15552000; preload" always;
    add_header Content-Security-Policy $policy always;
    add_header X-Content-Type-Options "nosniff" always;
    add_header X-Frame-Options "SAMEORIGIN" always;
    add_header X-XSS-Protection "1; mode=block" always;

    # Django API
    location /api {
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_redirect off;

        proxy_pass http://django-upstream;
    }

    # Health check
    location = /health-check/ {
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_redirect off;

        proxy_pass http://django-upstream;
    }

    # Static Assets
    location / {
        root /srv/dist;
        index index.html;

        try_files $uri $uri/ =404;
    }
}
```

I think most of what you have here for the virtual host configuration is the same. I just pasted this one so that there was a reference from where it begins and ends.